panther_gpiod/include/panther_gpiod/gpio_driver.hpp

```diff
@@ -0,0 +1,266 @@
+ * manages GPIO pins on the Panther robot by providing functionalities to set pin values,
+ * change pin directions, monitor pin events, and more.
+ */
+
+#ifndef PANTHER_GPIOD_GPIO_DRIVER_HPP_
+#define PANTHER_GPIOD_GPIO_DRIVER_HPP_
+
+#include <atomic>
+#include <filesystem>
+#include <functional>
+#include <map>
+#include <memory>
+#include <shared_mutex>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include <gpiod.hpp>
+
+namespace panther_gpiod
+{
+
+/**
+ * @brief Enumeration representing available GPIO pins in the Panther system.
+ */
+enum class GPIOPin {
+  AUX_PW_EN,
+  CHRG_DISABLE,
+  CHRG_SENSE,
+  DRIVER_EN,
+  E_STOP_RESET,
+  FAN_SW,
+  GPOUT1,
+  GPOUT2,
+  GPIN1,
+  GPIN2,
+  LED_SBC_SEL,
+  SHDN_INIT,
+  STAGE2_INPUT,
+  VDIG_OFF,
+  VMOT_ON,
+  WATCHDOG
+};
+
+/**
+ * @brief Structure containing information related to GPIO pins such as pin configuration,
+ * direction, value, etc. This information is required during the initialization process.
+ */
+struct GPIOInfo
+{
+  GPIOPin pin;
+  gpiod::line::direction direction;
+  bool active_low = false;
+  gpiod::line::value init_value = gpiod::line::value::INACTIVE;
+  gpiod::line::value value = gpiod::line::value::INACTIVE;
+  gpiod::line::offset offset = -1;
+};
+
+/**
+ * @brief Class responsible for managing GPIO pins on Panther robots, handling tasks such as
+ * setting pin values, changing pin directions, monitoring pin events, and more.
+ */
+class GPIODriver
+{
+public:
+  /**
+   * @brief Constructs the GPIODriver object with information about GPIO pin configurations.
+   * This information is necessary for initializing the GPIO functionality.
+   *
+   * @param gpio_info_storage Vector containing information about GPIO pin configurations.
+   * @param use_rt Whether to configure RT FIFO scheduling policy for the monitor thread.
+   *               Default is set to false.
+   * @param gpio_monit_thread_sched_priority Priority for the GPIO monitoring thread.
+   *        Set within the range of 0-99 to enable and configure the FIFO RT scheduling policy
+   *        for the monitor thread. This parameter is considered only if `use_rt` is set to true.
+   *        The default priority is 60.
+   *
+   * @par Example
+   * An example of constructing the GPIODriver object by providing GPIO pin information:
+   * @code{.cpp}
+   * std::vector<GPIOInfo> gpio_configurations = {
+   *   GPIOInfo{GPIOPin::CHRG_SENSE, gpiod::line::direction::INPUT},
+   *   GPIOInfo{GPIOPin::AUX_PW_EN, gpiod::line::direction::OUTPUT},
+   *   GPIOInfo{GPIOPin::LED_SBC_SEL, gpiod::line::direction::OUTPUT, true,
+   * gpiod::line::value::ACTIVE}
+   *   // ... additional GPIO pin configurations
+   * };
+   * GPIODriver gpio_driver(gpio_configurations);
+   * @endcode
+   */
+  GPIODriver(
+    std::vector<GPIOInfo> gpio_info_storage, const bool use_rt = false,
+    const int gpio_monit_thread_sched_priority = 60, const bool enable_gpio_monitoring = true);
```

Comment:
why did you add `enable_gpio_monitoring` parameter? To pass the value to it, first it will be necessary to pass RT parameters, which probably will be strange, as I think they are unrelated.